src/components/FormStep.js

```diff
@@ -481,16 +624,39 @@ const FormStep = ({form, submission, onLogicChecked, onStepSubmitted, onLogout})
     }
   };
 
+  /**
+   * Gets called when the user confirms saving the form.
+   * @return {Promise}
+   */
   const onSaveConfirm = async () => {
-    const response = await submitStepData(submissionStep.url, {...getCurrentFormData()});
-    return response;
+    return await submitStepData(submissionStep.url, {...getCurrentFormData()});
   };
 
+  /**
+   * Get called when the user presses the save button.
+   * @see {onSaveConfirm} Opens a modal, `onSaveConfirm` after user confirms.
+   *
+   * During evaluation, the following actions may be dispatched:
```

Comment:
seems to imply that sometimes it's dispatched, sometimes not, but it's always dispatched. & see earlier remarks about this documentation style for action dispatching -> I'd leave this out of the docblock.